### PR TITLE
Fix Issue: ExecEngine should use STRONG_TYPEDEF(col_id_t) #663

### PR DESCRIPTION
**Pull request by wgy8283335**
Fix Issue: ExecEngine should use STRONG_TYPEDEF(col_id_t) #663

### CODE REVIEW

**Comment by wgy8283335** [comment]
@apavlo . Hi, could you review my pr?

**Comment by wgy8283335** [comment]
Hi, my friend. Ci process always fail due to timeout, is there any wrong with Jenkins or travis? @mbutrovich

**Comment by apavlo** [comment]
@wgy8283335 Ignore the travis error. We need to disable it. The Jenkins error is another transient issue that we need to fix. I will restart your build.

File: src/include/execution/vm/bytecode_handlers.h [hunk 1]
```diff
@@ -136,6 +136,7 @@ INT_TYPES(BITS);
 // ---------------------------------------------------------
 // Memory operations
 // ---------------------------------------------------------
+typedef uint16_t col_id_t;
```
Comment (mbutrovich) [inline src/include/execution/vm/bytecode_handlers.h:139]: This isn't using `STRONG_TYPEDEF`. Please take a look at where `col_id_t` is already defined, and we'll actually need to refactor the signatures of functions that take this as an arg rather than letting it implicitly cast.

**Comment by wgy8283335** [comment]
I've run the unit test cases in local. All of them pass.
```
Test project /home/wangguangyuan/CLionProjects/terrier/cmake-build-debug
        Start   1: binder_test
  1/115 Test   #1: binder_test .........................   Passed    2.08 sec
        Start   2: catalog_test
  2/115 Test   #2: catalog_test ........................   Passed    0.85 sec
        Start   3: name_builder_test
  3/115 Test   #3: name_builder_test ...................   Passed    0.05 sec
        Start   4: bitmap_test
  4/115 Test   #4: bitmap_test .........................   Passed    3.91 sec
        Start   5: concurrent_bitmap_test
  5/115 Test   #5: concurrent_bitmap_test ..............   Passed    6.93 sec
        Start   6: hash_util_test
  6/115 Test   #6: hash_util_test ......................   Passed    0.08 sec
        Start   7: managed_pointer_test
  7/115 Test   #7: managed_pointer_test ................   Passed    0.06 sec
        Start   8: object_pool_test
  8/115 Test   #8: object_pool_test ....................   Passed    1.36 sec
        Start   9: perf_monitor_test
  9/115 Test   #9: perf_monitor_test ...................   Passed    4.13 sec
        Start  10: performance_counter_test
 10/115 Test  #10: performance_counter_test ............   Passed    0.65 sec
        Start  11: rusage_monitor_test
 11/115 Test  #11: rusage_monitor_test .................   Passed    5.40 sec
        Start  12: stat_registry_test
 12/115 Test  #12: stat_registry_test ..................   Passed    0.05 sec
        Start  13: worker_pool_test
 13/115 Test  #13: worker_pool_test ....................   Passed    2.07 sec
        Start  14: aggregation_hash_table_test
 14/115 Test  #14: aggregation_hash_table_test .........   Passed    0.82 sec
        Start  15: aggregators_test
 15/115 Test  #15: aggregators_test ....................   Passed    0.06 sec
        Start  16: arithmetic_functions_test
 16/115 Test  #16: arithmetic_functions_test ...........   Passed    0.06 sec
        Start  17: ast_context_test
 17/115 Test  #17: ast_context_test ....................   Passed    0.06 sec
        Start  18: ast_dump_test
 18/115 Test  #18: ast_dump_test .......................   Passed    0.07 sec
        Start  19: ast_test
 19/115 Test  #19: ast_test ............................   Passed    0.06 sec
        Start  20: ast_traversal_visitor_test
 20/115 Test  #20: ast_traversal_visitor_test ..........   Passed    0.07 sec
        Start  21: bandit_test
 21/115 Test  #21: bandit_test .........................   Passed    0.08 sec
        Start  22: bit_util_test
 22/115 Test  #22: bit_util_test .......................   Passed    0.06 sec
        Start  23: bitfield_test
 23/115 Test  #23: bitfield_test .......................   Passed    0.05 sec
        Start  24: bloom_filter_test
 24/115 Test  #24: bloom_filter_test ...................   Passed    2.62 sec
        Start  25: bytecode_generator_test
 25/115 Test  #25: bytecode_generator_test .............   Passed    0.23 sec
        Start  26: bytecode_iterator_test
 26/115 Test  #26: bytecode_iterator_test ..............   Passed    0.06 sec
        Start  27: bytecode_trampoline_test
 27/115 Test  #27: bytecode_trampoline_test ............   Passed    0.11 sec
        Start  28: bytecodes_test
 28/115 Test  #28: bytecodes_test ......................   Passed    0.07 sec
        Start  29: chunked_vector_test
 29/115 Test  #29: chunked_vector_test .................   Passed    0.92 sec
        Start  30: compiler_test
 30/115 Test  #30: compiler_test .......................   Passed    9.13 sec
        Start  31: concise_hash_table_test
 31/115 Test  #31: concise_hash_table_test .............   Passed    0.08 sec
        Start  32: date_functions_test
 32/115 Test  #32: date_functions_test .................   Passed    0.06 sec
        Start  33: ddl_executors_test
 33/115 Test  #33: ddl_executors_test ..................   Passed    0.88 sec
        Start  34: error_reporter_test
 34/115 Test  #34: error_reporter_test .................   Passed    0.05 sec
        Start  35: filter_manager_test
 35/115 Test  #35: filter_manager_test .................   Passed    0.13 sec
        Start  36: generic_hash_table_test
 36/115 Test  #36: generic_hash_table_test .............   Passed    0.13 sec
        Start  37: hash_test
 37/115 Test  #37: hash_test ...........................   Passed    0.08 sec
        Start  38: index_iterator_test
 38/115 Test  #38: index_iterator_test .................   Passed    1.62 sec
        Start  39: is_null_predicate_test
 39/115 Test  #39: is_null_predicate_test ..............   Passed    0.09 sec
        Start  40: join_hash_table_test
 40/115 Test  #40: join_hash_table_test ................   Passed    0.11 sec
        Start  41: join_hash_table_vector_probe_test
 41/115 Test  #41: join_hash_table_vector_probe_test ...   Passed    0.25 sec
        Start  42: math_util_test
 42/115 Test  #42: math_util_test ......................   Passed    0.05 sec
        Start  43: parsing_test
 43/115 Test  #43: parsing_test ........................   Passed    0.06 sec
        Start  44: pr_filler_test
 44/115 Test  #44: pr_filler_test ......................   Passed    0.42 sec
        Start  45: projected_columns_iterator_test
 45/115 Test  #45: projected_columns_iterator_test .....   Passed    0.76 sec
        Start  46: region_test
 46/115 Test  #46: region_test .........................   Passed    0.05 sec
        Start  47: scanner_test
 47/115 Test  #47: scanner_test ........................   Passed    0.06 sec
        Start  48: sema_builtin_test
 48/115 Test  #48: sema_builtin_test ...................   Passed    0.07 sec
        Start  49: sema_expr_test
 49/115 Test  #49: sema_expr_test ......................   Passed    0.07 sec
        Start  50: sorter_test
 50/115 Test  #50: sorter_test .........................   Passed    0.81 sec
        Start  51: storage_interface_test
 51/115 Test  #51: storage_interface_test ..............   Passed    1.40 sec
        Start  52: string_functions_test
 52/115 Test  #52: string_functions_test ...............   Passed    0.08 sec
        Start  53: table_vector_iterator_test
 53/115 Test  #53: table_vector_iterator_test ..........   Passed    1.15 sec
        Start  54: thread_state_container_test
 54/115 Test  #54: thread_state_container_test .........   Passed    0.42 sec
        Start  55: token_test
 55/115 Test  #55: token_test ..........................   Passed    0.12 sec
        Start  56: type_test
 56/115 Test  #56: type_test ...........................   Passed    0.07 sec
        Start  57: vector_util_test
 57/115 Test  #57: vector_util_test ....................   Passed    0.10 sec
        Start  58: tpcc_test
 58/115 Test  #58: tpcc_test ...........................   Passed    0.13 sec
        Start  59: metrics_test
 59/115 Test  #59: metrics_test ........................   Passed    6.29 sec
        Start  60: network_test
 60/115 Test  #60: network_test ........................   Passed    0.53 sec
        Start  61: postgres_protocol_util_test
 61/115 Test  #61: postgres_protocol_util_test .........   Passed    0.05 sec
        Start  62: column_stats_test
 62/115 Test  #62: column_stats_test ...................   Passed    0.05 sec
        Start  63: count_min_sketch_test
 63/115 Test  #63: count_min_sketch_test ...............   Passed    0.12 sec
        Start  64: histogram_test
 64/115 Test  #64: histogram_test ......................   Passed    0.13 sec
        Start  65: hyperloglog_test
 65/115 Test  #65: hyperloglog_test ....................   Passed    5.35 sec
        Start  66: logical_operator_test
 66/115 Test  #66: logical_operator_test ...............   Passed    0.30 sec
        Start  67: operator_transformer_test
 67/115 Test  #67: operator_transformer_test ...........   Passed    3.14 sec
        Start  68: optimizer_context_test
 68/115 Test  #68: optimizer_context_test ..............   Passed    0.06 sec
        Start  69: physical_operator_test
 69/115 Test  #69: physical_operator_test ..............   Passed    0.26 sec
        Start  70: stats_storage_test
 70/115 Test  #70: stats_storage_test ..................   Passed    0.05 sec
        Start  71: table_stats_test
 71/115 Test  #71: table_stats_test ....................   Passed    0.05 sec
        Start  72: top_k_elements_test
 72/115 Test  #72: top_k_elements_test .................   Passed    0.07 sec
        Start  73: tpcc_plan_delivery_test
 73/115 Test  #73: tpcc_plan_delivery_test .............   Passed    1.17 sec
        Start  74: tpcc_plan_index_scan
 74/115 Test  #74: tpcc_plan_index_scan ................   Passed    1.01 sec
        Start  75: tpcc_plan_neworder_test
 75/115 Test  #75: tpcc_plan_neworder_test .............   Passed    1.55 sec
        Start  76: tpcc_plan_orderstatus_test
 76/115 Test  #76: tpcc_plan_orderstatus_test ..........   Passed    0.67 sec
        Start  77: tpcc_plan_payment_test
 77/115 Test  #77: tpcc_plan_payment_test ..............   Passed    1.25 sec
        Start  78: tpcc_plan_seq_scan
 78/115 Test  #78: tpcc_plan_seq_scan ..................   Passed    0.85 sec
        Start  79: tpcc_plan_stocklevel_test
 79/115 Test  #79: tpcc_plan_stocklevel_test ...........   Passed    0.54 sec
        Start  80: value_condition_test
 80/115 Test  #80: value_condition_test ................   Passed    0.04 sec
        Start  81: expression_defs_test
 81/115 Test  #81: expression_defs_test ................   Passed    0.05 sec
        Start  82: expression_test
 82/115 Test  #82: expression_test .....................   Passed    0.17 sec
        Start  83: expression_util_test
 83/115 Test  #83: expression_util_test ................   Passed    0.04 sec
        Start  84: parser_test
 84/115 Test  #84: parser_test .........................   Passed    0.09 sec
        Start  85: pg_parser_test
 85/115 Test  #85: pg_parser_test ......................   Passed    0.06 sec
        Start  86: output_schema_test
 86/115 Test  #86: output_schema_test ..................   Passed    0.09 sec
        Start  87: plan_node_json_test
 87/115 Test  #87: plan_node_json_test .................   Passed    0.09 sec
        Start  88: plan_node_test
 88/115 Test  #88: plan_node_test ......................   Passed    0.06 sec
        Start  89: settings_test
 89/115 Test  #89: settings_test .......................   Passed    0.18 sec
        Start  90: access_observer_test
 90/115 Test  #90: access_observer_test ................   Passed    0.06 sec
        Start  91: block_access_controller_test
 91/115 Test  #91: block_access_controller_test ........   Passed    5.06 sec
        Start  92: block_compactor_test
 92/115 Test  #92: block_compactor_test ................   Passed    8.14 sec
        Start  93: bwtree_index_test
 93/115 Test  #93: bwtree_index_test ...................   Passed    4.73 sec
        Start  94: bwtree_test
 94/115 Test  #94: bwtree_test .........................   Passed  152.06 sec
        Start  95: data_table_concurrent_test
 95/115 Test  #95: data_table_concurrent_test ..........   Passed    3.82 sec
        Start  96: data_table_test
 96/115 Test  #96: data_table_test .....................   Passed    6.40 sec
        Start  97: delta_record_test
 97/115 Test  #97: delta_record_test ...................   Passed    1.25 sec
        Start  98: export_table_test
 98/115 Test  #98: export_table_test ...................***Failed    0.33 sec
        Start  99: garbage_collector_test
 99/115 Test  #99: garbage_collector_test ..............   Passed    1.04 sec
        Start 100: hash_index_test
100/115 Test #100: hash_index_test .....................   Passed    5.15 sec
        Start 101: index_key_test
101/115 Test #101: index_key_test ......................   Passed    0.85 sec
        Start 102: large_garbage_collector_test
102/115 Test #102: large_garbage_collector_test ........   Passed   59.07 sec
        Start 103: log_test
103/115 Test #103: log_test ............................   Passed    0.20 sec
        Start 104: projected_row_test
104/115 Test #104: projected_row_test ..................   Passed    1.81 sec
        Start 105: recovery_test
105/115 Test #105: recovery_test .......................   Passed    3.55 sec
        Start 106: storage_util_test
106/115 Test #106: storage_util_test ...................   Passed    3.62 sec
        Start 107: tuple_access_strategy_test
107/115 Test #107: tuple_access_strategy_test ..........   Passed   11.76 sec
        Start 108: varlen_entry_test
108/115 Test #108: varlen_entry_test ...................   Passed    0.04 sec
        Start 109: traffic_cop_test
109/115 Test #109: traffic_cop_test ....................   Passed    0.75 sec
        Start 110: deferred_actions_test
110/115 Test #110: deferred_actions_test ...............   Passed    0.05 sec
        Start 111: large_transaction_test
111/115 Test #111: large_transaction_test ..............   Passed   52.82 sec
        Start 112: mvcc_test
112/115 Test #112: mvcc_test ...........................   Passed    1.62 sec
        Start 113: transient_value_test
113/115 Test #113: transient_value_test ................   Passed    0.32 sec
        Start 114: type_util_test
114/115 Test #114: type_util_test ......................   Passed    0.04 sec
        Start 115: time_util_test
115/115 Test #115: time_util_test ......................   Passed    0.05 sec

99% tests passed, 1 tests failed out of 115
```

File: src/include/execution/vm/bytecode_handlers.h [hunk 2]
```diff
@@ -34,6 +34,11 @@
 #define VM_OP_WARM VM_OP inline
 #define VM_OP_COLD VM_OP NEVER_INLINE
 
+namespace terrier::execution::vm {
+STRONG_TYPEDEF(col_id_t, uint16_t);
```
Comment (mbutrovich) [inline src/include/execution/vm/bytecode_handlers.h:38]: This is already defined in `include/storage/storage_defs.h`. We don't want another definition in a different namespace.

File: src/include/execution/vm/bytecode_handlers.h [hunk 3]
```diff
@@ -34,6 +34,11 @@
...
+STRONG_TYPEDEF(col_id_32_t, uint32_t);
```
Comment (mbutrovich) [inline src/include/execution/vm/bytecode_handlers.h:39]: Shouldn't need a separate type for this. We can cast as needed.

**Comment by wgy8283335** [comment]
@mbutrovich hello, ci still has problems, and the pull request fails to merge into the master branch.

File: src/include/execution/vm/bytecode_handlers.h (hunk 2 above)
Comment (wgy8283335) [reply src/include/execution/vm/bytecode_handlers.h]: I have fix this.

File: src/include/execution/vm/bytecode_handlers.h (hunk 3 above)
Comment (wgy8283335) [reply src/include/execution/vm/bytecode_handlers.h]: I have fix this.

**Comment by wgy8283335** [comment]
@apavlo hello. Does CI still have problems?  The pr is blocked by the problem.

**Comment by apavlo** [comment]
@butro Can you please approve/merge this PR?

File: src/execution/vm/vm.cpp [hunk 4]
```diff
@@ -660,14 +660,16 @@ void VM::Interpret(const uint8_t *ip, Frame *frame) {
     auto *result = frame->LocalAt<type *>(READ_LOCAL_ID());                       \
     auto *pci = frame->LocalAt<sql::ProjectedColumnsIterator *>(READ_LOCAL_ID()); \
     auto col_idx = READ_UIMM2();                                                  \
-    OpPCIGet##type_str(result, pci, col_idx);                                     \
+    terrier::storage::col_id_t col_id(col_idx);                                   \
```
Comment (pmenon) [inline src/execution/vm/vm.cpp:663]: Undo these changes in the VM.

File: src/include/execution/vm/bytecode_handlers.h [hunk 5]
```diff
@@ -292,9 +292,10 @@ VM_OP_HOT void OpPCIGetSlot(terrier::storage::TupleSlot *slot, terrier::executio
 }
 
 VM_OP_HOT void OpPCIGetBool(terrier::execution::sql::BoolVal *out,
-                            terrier::execution::sql::ProjectedColumnsIterator *iter, uint16_t col_idx) {
+                            terrier::execution::sql::ProjectedColumnsIterator *iter,
```
Comment (pmenon) [inline src/include/execution/vm/bytecode_handlers.h:295]: Undo the changes to the handler signatures.

File: src/include/execution/vm/bytecode_handlers.h [hunk 6]
```diff
@@ -292,9 +292,10 @@ VM_OP_HOT void OpPCIGetSlot(terrier::storage::TupleSlot *slot, terrier::executio
...
+                            terrier::storage::col_id_t col_idx) {
   // Read
-  auto *ptr = iter->Get<bool, false>(col_idx, nullptr);
+  auto *ptr = iter->Get<bool, false>(!col_idx, nullptr);
```
Comment (pmenon) [inline src/include/execution/vm/bytecode_handlers.h:298]: Modify `terrier::execution::sql::ProjectedColumnsIterator::Get` to accept a `terrier::storage::col_id_t`. Perform the conversion from primitive to strong typedef here.

**Comment by wgy8283335** [comment]
One of the unit tests failed in Jenkins.
```
CLionProjects/terrier/cmake-build-debug/debug/export_table_test --gtest_filter=* --gtest_color=no
Running main() from gmock_main.cc
Traceback (most recent call last):
  File "transform_table.py", line 1, in <module>
    import pyarrow as pa
ModuleNotFoundError: No module named 'pyarrow'
```
Does the pyarrow exist in the Jenkins?
@pmenon

**Comment by wgy8283335** [comment]
run ci.

**Comment by apavlo** [comment]
Closed to reduce Jenkins contention
